### DISCUSSION TITLE
Suricata Update

### DISCUSSION THREAD

**Discussion by gFazzari** [jasonish/docker-suricata#31] [General]
Hi!

I'd like to know if there is a way to run suricata-update inside the container.
I tried to extend the base image installing the crontab package, but I think the crond process is not running.
Do you have any advice to make it work?

Thanks.

**Comment by jasonish** [comment]
Suricata-Update can run inside the container, but you still have to execute on your own.  Just adding cron to the container is not enough.

Say you have Suricata running in a container:
```
docker run --rm -it --privileged --name=suricata --net=host jasonish/suricata:latest -i enp10s0 -vvv
```
from outside the container you can do something like:
```
docker exec -it suricata suricata-update
```
If watching the output of the Suricata container you will see that `suricata-update` does trigger a rule reload.

Cron is already installed, so you could extend the image, add the relevant cron files to run Suriata-Update and then start cron after your Suricata container is started:
```
docker exec suricata crond
```

But this isn't provided out of the box. Its up to the user to work into their own container orchestration, at least for now.

**Comment by j0nny55555** [comment]
Just started to use your docker container to more or less do this, I also made a 'dummy' suricatamin.yaml so suricata is more or less doing nothing and while the container is running, exec the suricata-update, then copy the rule file out for distribution

Did it as a docker-compose.yml and so I `docker compose up -d`, then run the `docker exec ...` update command, `docker compose down`, and all done!

Was able to enable all the rule sources and have volumes set for the correct paths to ensure state sticks around - thank you very much @jasonish ^_^

My question is, are there any other files besides the new rules file that is created that needs to be copied to the host running that rules file?

**Reply by jasonish** [reply]
You should consider the full content of the `/var/lib/suricata/rules` directory.  While the ET rules might not be using datasets, `pawpatrules` is, so this will result in some extra files and directories under that directory.

**Reply by j0nny55555** [reply]
With pawpatrules enabled, and when `suricata-update` gets done, there are two files in the `/var/lib/suricata/rules` folder - the rules file and a classification.config file

It would seem fortunately there are no additional files, and I've already modified my classification to enable banning the way I want via CrowdSec (1 - major, 2 - medium, 3 - ignored)

**Reply by jasonish** [reply]
My test is...
```
cd ~/scratch
rm -rf *
docker run --rm -it -v $(pwd)/lib:/var/lib/suricata jasonish/suricata:latest suricata-update enable-source pawpatrules
docker run --rm -it -v $(pwd)/lib:/var/lib/suricata jasonish/suricata:latest suricata-update
```
Then...
```
find lib/ -type f
```
Results in:
```
lib/update/sources/et-open.yaml
lib/update/sources/pawpatrules.yaml
lib/update/cache/0d24534050aaf2021175379aba13eeb3-paw-patrules.tar.gz
lib/update/cache/04e5c82359d9d1302844e72993296635-emerging.rules.tar.gz
lib/rules/datasets/789a2b835a8844296efb9a037523d3bf
lib/rules/datasets/3e6c5358aa5d6f6d3a0887e3b4cd0814
lib/rules/datasets/eeab26fc22636fd972976e118d196f8f
lib/rules/datasets/34c3ba3122e512820052a1f2ba1e6c2f
lib/rules/datasets/8bbcff8e1b540c8e20b65d20a44a3252
lib/rules/datasets/ea8ac6bf247f409e8ef8a0abd07cdf37
lib/rules/datasets/1cfec84d82bdb0d4c44ce4767a79f211
lib/rules/datasets/fc7f1bf3299c2df56456a195f07f08f9
lib/rules/datasets/ebc5c6abfc73569a2a3778803b538738
lib/rules/datasets/60f0a25e6ae1dd3376b68840cc16ce22
lib/rules/datasets/fa6b492d74837057bddce3ca8bdb1774
lib/rules/datasets/7525eeb78c73dadb502f26e6d435ce80
lib/rules/datasets/d7fa85b0bbb522ff2fe857ff25e0ba78
lib/rules/datasets/aba47376d20c6b1c9aa9daf84942c31c
lib/rules/datasets/adcf3efb23307665b749424ab64825fc
lib/rules/datasets/e01ec07296517c72dae5443f651dc1e3
lib/rules/datasets/1a7599892809ff05e1230f5ea8b4b798
lib/rules/classification.config
lib/rules/suricata.rules
```
you should see a handful of dataset files in the datasets directory.
